File: src/services/scraper.js
import { parseHTML } from '../utils/htmlParser.js';

export async function scrapeUrl(url) {
  try {
    const response = await fetch(url);
    const html = await response.text();
    return parseHTML(html);
  } catch (error) {
    throw new Error(`Failed to scrape URL: ${error.message}`);
  }
}

File: src/scraper/scraper.py
import requests
from .html_parser import HTMLParser

class WebScraper:
    def __init__(self):
        self.parser = HTMLParser()
    
    def scrape(self, url):
        try:
            response = requests.get(url)
            response.raise_for_status()
            return self.parser.parse(response.text)
        except requests.RequestException as e:
            return {'error': str(e)}